.github/workflows/docs.yml

```diff
@@ -0,0 +1,46 @@
+name: "Docs"
+
+concurrency:
+  group: docs
+  cancel-in-progress: true
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v2
+      with:
+        path: src
+
+      # The checkout action ensures the correct auth tokens
+      # are set in the repo so we can push. We will pass
+      # this path to the deploy stage to use instead of
+      # its own temporary checkout.
+    - name: Checkout docs branch
+      uses: actions/checkout@v2
+      with:
+        path: target
+        ref: docs
+
+    - name: Build
+      run: |
+        # TODO [tc] We should really publish the Docker
```

Comment:
[plan] Indeed! Testing this on my fork and it's taking a looong time to run.

Reply:
Added https://github.com/TheFoundryVisionmongers/OpenAssetIO/issues/109